gensim/models/experimental/drmm_tks.py

```diff
@@ -0,0 +1,595 @@
+
+try:
+    import keras.backend as K
+    from keras import optimizers
+    from keras.losses import hinge
+    from keras.models import Model
+    from keras.layers import Input, Embedding, Dot, Dense, Reshape, Dropout
+    import tensorflow
+    tensorflow.set_random_seed(101010)
+    KERAS_AVAILABLE = True
+except ImportError:
+    KERAS_AVAILABLE = False
+
+random.seed(101010)
+numpy.random.seed(101010)
+logger = logging.getLogger(__name__)
+
+
+class DRMM_TKS(utils.SaveLoad):
+    """User friendly model for training on similarity learning data.
+    You only have to provide sentences in the data as a list of words.
+
+    Example Usage:
+    -------------
+    drmm_tks_model = DRMM_TKS_Model(queries, docs, labels, word_embedding_path)
+    drmm_tks_model.predict(test_queries, test_docs)
+
+    The data should have the format:
+    queries = ["When was World Wat 1 fought ?".split(),
+             "When was Gandhi born ?".split()]
+
+    docs = [
+            ["The world war was bad".split(),
+            "It was fought in 1996".split()],
+            ["Gandhi was born in the 18th century".split(),
+             "He fought for the Indian freedom movement".split(),
+             "Gandhi was assasinated".split()]
+           ]
+
+    labels = [[0, 1],
+              [1, 0, 0]]
+    """
+
+    def __init__(self, queries, docs, labels, word_embedding_path=None,
+                 text_maxlen=200, keep_full_embedding=True, normalize_embeddings=True,
+                 epochs=10, unk_handle_method='zero', validation_data=None):
+        """Initializes the model and trains it
+
+        Parameters:
+        -----------
+        queries: list of list of string words
+            The questions for the similarity learning model
+            Example:
+            queries=["When was World Wat 1 fought ?".split(),
+                     "When was Gandhi born ?".split()],
+
+        docs: list of list of list of string words
+            The candidate answers for the similarity learning model
+
+            Example:
+            docs = [
+                    ["The world war was bad".split(),
+                    "It was fought in 1996".split()],
+                    ["Gandhi was born in the 18th century".split(),
+                     "He fought for the Indian freedom movement".split(),
+                     "Gandhi was assasinated".split()]
+                   ]
+
+        labels: list of list of ints
+            Indicates when a candidate document is relevant to a query
+            1 : relevant
+            0 : irrelevant
+
+            Example:
+            labels = [[0, 1],
+                      [1, 0, 0]]
+
+        word_embedding_path: str
+            path to the Glove vectors which have the embeddings in a .txt format
+            If unset, random word embeddings will be used
+
+        text_maxlen: int
+            The maximum possible length of a query or a document
+            This is used for padding.
+
+        keep_full_embedding: boolean
```

Comment:
honestly, useless parameter, better to keep full always.

Reply:
What about when the user has a very big embedding file (say `Glove 840B` tokens)
and can't keep all the words in memory.

if user sets keep full embedding to True,
the Embedding layer has:
1. words in his corpus
2. words in the embedding file but not in his corpus.

If we always `set keep_full_embedding` to True, 
He might get too many words in the embedding file but not in his corpus, which he might not need.

What do you think?

Reply:
By my experience, better to have it always (because trainset anyway will be small and don't cover well all needed words).
If the user has really large embeddings (like you suggest), he can reconstruct it and pass "compressed" kv to your code (with only needed vectors, that's possible).